commit 8efef12eeb377e9087551e333d65631968c80e5f
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat Jun 10 17:00:25 2017 +0200

    Fix build with future glibc (major/minor macros), BSD compatible
    
    Summary:
    Also adding sys/types.h which is not implicitly included from all over.
    
    Thanks-to: [ade] in #kde-devel
    
    See also:
    https://sourceware.org/ml/libc-alpha/2015-11/msg00452.html
    https://git.reviewboard.kde.org/r/127662/
    
    Reviewers: #plasma
    
    Subscribers: plasma-devel, kwin, #kwin, adridg
    
    Tags: #kwin
    
    Differential Revision: https://phabricator.kde.org/D6175

diff --git a/logind.cpp b/logind.cpp
index fee0f2ff4..bb84af4f9 100644
--- a/logind.cpp
+++ b/logind.cpp
@@ -29,6 +29,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <QDBusMetaType>
 
 #include <sys/stat.h>
+#include <sys/types.h>
+#ifndef major
+#include <sys/sysmacros.h>
+#endif
 #include <unistd.h>
 #include "utils.h"
 
